### PR TITLE
add ArgMarshaler for package@version args

### PR DESCRIPTION
**Pull request by mdrohmann**
https://www.pivotaltracker.com/story/show/176194323

### CODE REVIEW

**Comment by mdrohmann** [comment]
This unifies the behavior for arguments of `language@version`, `platform@version`, `package@version` and `channel@version`.

File: internal/captain/nameversion.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+}
+
+func (nv *NameVersion) Set(arg string) error {
+	nameArg := strings.Split(arg, "@")
+	nv.name = nameArg[0]
+	if len(nameArg) == 2 {
+		nv.version = nameArg[1]
+	}
+	if len(nameArg) > 2 {
+		return errors.New("invalid format")
```
Comment (MDrakos) [inline internal/captain/nameversion.go:23]: I think this needs localization and also could be a more descriptive error

File: internal/config/config_test.go [hunk 2]
```diff
@@ -108,6 +108,7 @@ func (suite *ConfigTestSuite) testNoHomeRunner() {
 }
 
 func (suite *ConfigTestSuite) TestNoHome() {
+	suite.T().Skip()
```
Comment (MDrakos) [inline internal/config/config_test.go:111]: Was this for debugging?

**Review by MDrakos — CHANGES_REQUESTED** [review]
I wonder if `captain` is the right place for `nameversion.go`. That or maybe the types are too specific to our commands? I'm not sure if it's appropriate to give `captain` knowledge of what a `PackageVersion` is. Though I will defer to @Naatan for his thoughts.

Other than that I like how it's used in the runners and only noted two minor comments to address.

File: internal/config/config_test.go (hunk 2 above)
Comment (mdrohmann) [reply internal/config/config_test.go]: Thanks for catching!

**Comment by mdrohmann** [comment]
I think you are right about the location for the `NameVersion` specializations.  I moved the specializations to their respective runners.

**Review by MDrakos — DISMISSED** [review]

File: internal/runners/packages/packages.go [hunk 3]
```diff
@@ -21,6 +22,18 @@ import (
...
 )
 
+type PackageVersion struct {
+	captain.NameVersion
+}
+
+func (pv *PackageVersion) Set(arg string) error {
+	err := pv.NameVersion.Set(arg)
+	if err != nil {
+		return locale.NewError("err_package_format", "The package and version provided is not formatting correctly, must be in the form of <package>@<version>")
```
Comment (Naatan) [inline internal/runners/packages/packages.go:32]: Should be an input error

File: internal/captain/nameversion.go [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+func (nv *NameVersion) Set(arg string) error {
+	nameArg := strings.Split(arg, "@")
+	nv.name = nameArg[0]
+	if len(nameArg) == 2 {
+		nv.version = nameArg[1]
+	}
+	if len(nameArg) > 2 {
+		return locale.NewError("name_version_format_err", "Invalid format: Should be <name@version>")
```
Comment (Naatan) [inline internal/captain/nameversion.go:22]: Should be input error

File: internal/runners/platforms/platforms.go [hunk 5]
```diff
@@ -7,11 +7,24 @@ import (
...
 )
 
+type PlatformVersion struct {
+	captain.NameVersion
+}
+
+func (pv *PlatformVersion) Set(arg string) error {
+	err := pv.NameVersion.Set(arg)
+	if err != nil {
+		return locale.NewError("err_platform_format", "The platform and version provided is not formatting correctly, must be in the form of <platform>@<version>")
```
Comment (Naatan) [inline internal/runners/platforms/platforms.go:23]: Input error

**Review by Naatan — CHANGES_REQUESTED** [review]

**Review by Naatan — CHANGES_REQUESTED** [review]
You're also loosing the original error I just noticed. Should be WrapInputError not NewInputError.

**Review by Naatan — APPROVED** [review]

**Pull request merged**